### PR TITLE
Basic Structure

### PR DESCRIPTION
**Pull request by bryantgillespie**

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,2 @@
+DIRECTUS_URL="https://your-instance.directus.appp"
```
Comment (rijkvanzanten) [inline .env.example:1]:
```suggestion
DIRECTUS_URL="https://your-instance.directus.app"
```

File: .env.example [hunk 2]
```diff
@@ -0,0 +1,2 @@
...
+DIRECTUS_ADMIN_TOKEN="XHdfaklsdj;fladsjkfal;zj;lkda"
```
Comment (rijkvanzanten) [inline .env.example:2]: Do we need an _admin_ token? Do we need a token at all?

File: components/blocks/BlockComponentName.vue [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<script setup lang="ts">
+type Block = {
+	id: string;
+};
```
Comment (rijkvanzanten) [inline components/blocks/BlockComponentName.vue:4]: Use interfaces for object structure typing

```suggestion
interface Block {
	id: string;
}
```

File: components/blocks/BlockComponentName.vue [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+const props = defineProps({
+	data: {
+		type: Object as PropType<Block>,
+		required: true,
+	},
+});
```
Comment (rijkvanzanten) [inline components/blocks/BlockComponentName.vue:10]: Lets use TS to define props

```suggestion
interface BlockComponentNameProps {
  data: Block;
}

defineProps<BlockComponentNameProps>();
```

File: layouts/default.vue [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<script setup lang="ts"></script>
+<template>
+	<div>
+		<TheHeader />
+		<slot />
+		<TheFooter />
+	</div>
+</template>
```
Comment (rijkvanzanten) [inline layouts/default.vue:8]: Are those root elements still required in Nuxt?

```suggestion
<template>
	<TheHeader />
	<slot />
	<TheFooter />
</template>
```

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

File: .env.example (hunk 2 above)
Comment (bryantgillespie) [reply .env.example:2]: Doesn't have to be an _admin_ token I guess.

File: types.ts [hunk 6]
```diff
@@ -0,0 +1,28 @@
+// Page builder types
```
[the remark this answers is not in the document]
Comment (bryantgillespie) [reply types.ts:1]: So would all these just be interfaces?

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**